yarn-project/builder/src/contract-interface-gen/typescript.ts

```diff
@@ -43,6 +44,11 @@ function abiTypeToTypescript(type: ABIParameter['type']): string {
       if (isWrappedFieldStruct(type)) {
         return 'WrappedFieldLike';
       }
+      if (isBoundedVecStruct(type)) {
+        // To make BoundedVec easier to work with, we expect a simple array on the input and then we encode it
+        // as a BoundedVec in the ArgumentsEncoder.
+        return `${abiTypeToTypescript(type.fields[0].type)}`;
```

Comment:
This piece of code is generating the TS arguments types in the contract class. Had to do this change such that it expects an array.

Reply:
This means that the maximum array length is not present on the interface no? It'd fail at runtime when the call gets encoded and it finds an array that is too large.

Reply:
zomg you did not fail at runtime!

Reply:
Correct. I figured out a way to do it in [this commit](https://github.com/AztecProtocol/aztec-packages/pull/14891/commits/94ca29baeeb299a9ef65264023cf667c4c652c91) but since this would be a larger change I would like your feedback before going down that route.